server/src/main/java/org/opensearch/ratelimitting/admissioncontrol/controllers/IoBasedAdmissionController.java

```diff
@@ -0,0 +1,136 @@
+     * @param resourceUsageCollectorService instance used to get resource usage stats of the node
+     * @param clusterService                instance of the clusterService
+     */
+    public IoBasedAdmissionController(
+        String admissionControllerName,
+        ResourceUsageCollectorService resourceUsageCollectorService,
+        ClusterService clusterService,
+        Settings settings
+    ) {
+        super(admissionControllerName, resourceUsageCollectorService, clusterService);
+        this.settings = new IoBasedAdmissionControllerSettings(clusterService.getClusterSettings(), settings);
+    }
+
+    @Override
+    public boolean isEnabledForTransportLayer(AdmissionControlMode admissionControlMode) {
+        if (Constants.LINUX) {
+            return super.isEnabledForTransportLayer(admissionControlMode);
+        }
+        return false;
+    }
+
+    /**
+     * Apply admission control based on the resource usage for an action
+     *
+     * @param action is the transport action
+     * @param admissionControlActionType type of admissionControlActionType
+     */
+    @Override
+    public void apply(String action, AdmissionControlActionType admissionControlActionType) {
+        if (this.isEnabledForTransportLayer(this.settings.getTransportLayerAdmissionControllerMode())) {
+            this.applyForTransportLayer(action, admissionControlActionType);
+        }
+    }
+
+    /**
+     * Apply transport layer admission control if configured limit has been reached
+     */
+    private void applyForTransportLayer(String actionName, AdmissionControlActionType admissionControlActionType) {
+        if (isLimitsBreached(actionName, admissionControlActionType)) {
+            this.addRejectionCount(admissionControlActionType.getType(), 1);
+            if (this.isAdmissionControllerEnforced(this.settings.getTransportLayerAdmissionControllerMode())) {
+                throw new OpenSearchRejectedExecutionException(
+                    String.format(
+                        Locale.ROOT,
+                        "Io usage admission controller rejected the request for action [%s] as IO limit reached",
+                        admissionControlActionType.name()
+                    )
+                );
+            }
+        }
+    }
+
+    /**
+     * Check if the configured resource usage limits are breached for the action
+     */
+    private boolean isLimitsBreached(String actionName, AdmissionControlActionType admissionControlActionType) {
+        // check if cluster state is ready
+        if (clusterService.state() != null && clusterService.state().nodes() != null) {
+            long ioUsageThreshold = this.getIoRejectionThreshold(admissionControlActionType);
+            Optional<NodeResourceUsageStats> nodePerformanceStatistics = this.resourceUsageCollectorService.getNodeStatistics(
+                this.clusterService.state().nodes().getLocalNodeId()
```

Comment:
Lets store local node id in class member as well so we dont need to initialize this again and again

Reply:
As this value is get updated based on the trackers window duration we are not setting this by default.